### PR TITLE
Document the automated release process

### PR DESCRIPTION
**Pull request by felixfontein**
Based on comments of @gotmax23 in the #release-management:ansible.com room.

### CODE REVIEW

File: docs/automated-process.md
Comment (webknjaz) [inline docs/automated-process.md:17]: Double checkout is confusing here. Let's rephrase

Comment (felixfontein) [reply docs/automated-process.md:17]: Like this?
```suggestion
2. Check out the PR in your `ansible-build-data` clone and copy the updated porting guide into the `ansible-documentation` repository.
```

File: docs/automated-process.md
Comment (felixfontein) [inline docs/automated-process.md:14]:
```suggestion
1. Trigger [the automated workflow on GHA](https://github.com/ansible-community/ansible-build-data/actions/workflows/ansible-release.yml) and fill in the release version (like 11.2.0) and the major version (like 11).
```

**Comment by felixfontein** [comment]
The current preview of the new page can be seen here: https://ansible--381.org.readthedocs.build/projects/ansible-build-data/automated-process/ (link valid as long as PR is open)

File: docs/automated-process.md
Comment (felixfontein) [inline docs/automated-process.md:20]:
```suggestion
3. Once both PRs (in `ansible-build-data` and `ansible-documentation`) are approved, merge the `ansible-build-data` PR and approve the next workflow step (**in this order!** the next steps of the workflow require the PR to be merged!).
```

File: docs/automated-process.md
Comment (acozine) [inline docs/automated-process.md:5]: I'd add a link here to the docs for the manual process.

**Comment by felixfontein** [comment]
Maybe we should already move the "Create porting guide PR" to after merging the ansible-build-data PR? This will be necessary for the automated porting guide PR workflow that's currently under construction (https://github.com/ansible/ansible-documentation/pull/121).

File: docs/automated-process.md
Comment (acozine) [inline docs/automated-process.md:1]: We could make this a more action-oriented title. Maybe something like:
```suggestion
# How to release a new version of the Ansible Community Package - Automated Release Process
```
That gets wordy, though.

File: docs/automated-process.md
Comment (acozine) [inline docs/automated-process.md:15]: How/when does the approval happen?

File: docs/automated-process.md
Comment (acozine) [inline docs/automated-process.md:17]: Depending on the audience, you might want to fill in details - for example, the path to the updated porting guide, the path where it should go in the ansible-documentation repo.

File: docs/automated-process.md
Comment (acozine) [inline docs/automated-process.md:30]:
```suggestion
5. Announce the release on the Forum, Matrix, and the mailing list by running the following command in the `${MAJOR_VERSION}` directory of the `ansible-build-data` checkout:
    ```
    antsibull-build announcements --send --data-dir . ${VERSION}
    ```
```

File: docs/release-process.md
Comment (acozine) [inline docs/release-process.md:5]: I'd add a link here to the automated docs.

File: docs/release-process.md
Comment (acozine) [inline docs/release-process.md:148]: I would make this match the entry in the automated docs. GitHub won't let me make a suggestion on the unchanged line 144, though.

**Review by acozine — COMMENTED** [review]
This is awesome, thanks @felixfontein!

I added a few suggestions.

File: docs/automated-process.md
Comment (oraNod) [inline docs/automated-process.md:9]:
```suggestion
    `${PLACEHOLDER_VALUE}` where `PLACEHOLDER_VALUE` describes the value to specify.
```

Comment (oraNod) [reply docs/automated-process.md:9]: "Fill in" might be a little idiomatic. Prepositional verbs can be harder to understand.

File: docs/automated-process.md
Comment (oraNod) [inline docs/automated-process.md:14]:
```suggestion
1. Trigger [the automated workflow](https://github.com/ansible-community/ansible-build-data/actions/workflows/ansible-release.yml) on the **Actions** tab of the repository and specify the release version, such as 11.2.0, and the major version, such as 11.
```

File: docs/automated-process.md
Comment (oraNod) [inline docs/automated-process.md:15]: I believe this should be two sentences to avoid a comma splice.

File: docs/automated-process.md
Comment (oraNod) [inline docs/automated-process.md:20]:
```suggestion
3. After both PRs (in `ansible-build-data` and `ansible-documentation`) are approved, merge the `ansible-build-data` PR and approve the next workflow step (**in this order!** the next steps of the workflow require the PR to be merged!).
```

File: docs/automated-process.md
Comment (oraNod) [inline docs/automated-process.md:23]:
```suggestion
4. Merge the porting guide PR, and backport it to the latest `stable-x` branches down to the ansible-core version that is included in the Ansible release.
```

**Review by oraNod — COMMENTED** [review]
Few suggestions. Feel free to take them or leave them.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md:5]: Inserted in 5ad0ad71749cb2907b8cdd6115246bbd5d9b8045.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md]: Changed in 5ad0ad71749cb2907b8cdd6115246bbd5d9b8045.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md]: I've explained it in 5ad0ad71749cb2907b8cdd6115246bbd5d9b8045.

File: docs/release-process.md
Comment (felixfontein) [reply docs/release-process.md]: Adjusted in 5ad0ad71749cb2907b8cdd6115246bbd5d9b8045. I also adjusted this document's title similarly to the automated one.

File: docs/release-process.md
Comment (felixfontein) [reply docs/release-process.md:5]: Also done in 5ad0ad71749cb2907b8cdd6115246bbd5d9b8045.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md]: :+1: adffd2bdf9fcf79bb689070cb6fff2706cac8eb6.

**Comment by felixfontein** [comment]
Are there any more comments on this? Or should I merge the current version?

**Comment by gotmax23** [comment]
I'll try to give it a look later today.

**Review by oraNod — APPROVED** [review]
LGTM. Thank you @felixfontein

File: docs/release-process.md
Comment (gotmax23) [inline docs/release-process.md:1]:
```suggestion
# How to release a new version of the Ansible Community Package — Manual Release Process
```

File: docs/automated-process.md
Comment (gotmax23) [inline docs/automated-process.md:1]:
```suggestion
# How to release a new version of the Ansible Community Package — Automated Release Process
```

File: docs/automated-process.md
Comment (gotmax23) [inline docs/automated-process.md:17]:
```suggestion
   Afterwards, it will wait for approval before continuing with uploading the package to PyPI.
```

File: docs/automated-process.md
Comment (gotmax23) [inline docs/automated-process.md:18]: I believe only @ansible-community/release-management-wg is notified.

File: docs/automated-process.md
Comment (gotmax23) [inline docs/automated-process.md:1]: See https://github.com/ansible-community/ansible-build-data/blob/main/docs/CONVENTIONS.md. The Semantic Line Breaks standard says

> A maximum line length of 80 characters is RECOMMENDED.

which I have applied to the rest of the files in the `docs` directory.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md:18]: Why do you think so? That group isn't mentioned in the workflow, so I would assume that GH notifies everyone with appropriate rights, which probably is the set of folks with commit rights.

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md:1]: 7fdf5aab2beb02571c53a6cbd1d70c140c388270

File: docs/automated-process.md
Comment (acozine) [reply docs/automated-process.md]: Looks great, thanks @felixfontein.

File: docs/automated-process.md
Comment (gotmax23) [reply docs/automated-process.md]: I believe that's how the environment that @anweshadas created is set up. It's set in the repo settings, not the workflow file.

Comment (felixfontein) [reply docs/automated-process.md]: @anweshadas can you confirm this?

Comment (webknjaz) [reply docs/automated-process.md]: > Why do you think so? That group isn't mentioned in the workflow, so I would assume that GH notifies everyone with appropriate rights, which probably is the set of folks with commit rights.

In GitHub environments, there's protection rules, of which one is "required reviewers". These are listed as GitHub usernames or teams. The maximum is 6 entries. Whoever is listed will be notified, I don't think this will notify accounts that aren't in those teams or lists.

**Comment by felixfontein** [comment]
In case #390 gets merged first the process can be simplified a bit. (If this one gets merged first, the docs need up be updated in #390.)

**Comment by felixfontein** [comment]
I still don't know how to adjust https://github.com/ansible-community/ansible-build-data/pull/381#discussion_r1562798345 since I do not have access to the repository settings. @anweshadas or anyone else with sufficient rights, can you please check out which names and/or groups are listed as "Required reviewers" for the second workflow step, and how exactly the setting looks like (so we can describe it in the documentation - will make it easier to later validate for someone with appropriate access who didn't set this up)?

File: docs/automated-process.md
Comment (felixfontein) [reply docs/automated-process.md]: I don't have any way to verify this (I don't have access to the settings), so can someone with appropriate access please check this and confirm (or say what else it is)?

Comment (webknjaz) [reply docs/automated-process.md]: I don't have enough privileges in this repo to check this :(

Maybe @oraNod does?

Comment (oraNod) [reply docs/automated-process.md]: Hi, plz excuse the delayed response. Just getting back from PTO...

It looks like @gotmax23 is correct and only folks in the `ansible-community/release-management-wg` group are notified. Here's a screenshot of the required reviewers section of the environment:

image

On a side note, I'm surprised that @felixfontein and @webknjaz don't have sufficient repo permissions. Should we address that?

Comment (felixfontein) [reply docs/automated-process.md]: Thanks a lot! I've updated the paragraph in 81cc67840cdc38bf8b9e95557961b727c190dce4 and added a footnote explaining how this is configured.

Regarding repo permissions: I wouldn't mind having more permissions :) I have no idea who currently has more permissions and why the permissions are as they are...

**Comment by felixfontein** [comment]
If nobody objects, I'm going to merge this soon. The current preview of the new page is here: https://ansible--381.org.readthedocs.build/projects/ansible-build-data/automated-process/

**Comment by felixfontein** [comment]
Thanks everyone for reviewing and figuring things out :)

**Comment by oraNod** [comment]
Thanks @felixfontein !

**Pull request merged**
